dataactbroker/handlers/fileHandler.py

```diff
@@ -686,149 +679,211 @@ def publish_fabs_submission(submission):
-                            row_id = check_row.published_award_financial_assistance_id
-                            sess.query(pafa).\
-                                filter_by(published_award_financial_assistance_id=row_id).\
-                                update({'is_active': False, 'updated_at': row.modified_at}, synchronize_session=False)
-
-                    # for all rows, insert the new row (active/inactive should be handled by fabs_derivations)
-                    new_row = PublishedAwardFinancialAssistance(**temp_obj)
-                    sess.add(new_row)
-
-                    # update the list of affected awarding_agency_codes
-                    if temp_obj['awarding_agency_code'] not in agency_codes_list:
-                        agency_codes_list.append(temp_obj['awarding_agency_code'])
-
-                    # update the list of affected funding_agency_codes
-                    if temp_obj['funding_agency_code'] not in agency_codes_list:
-                        agency_codes_list.append(temp_obj['funding_agency_code'])
-
-                    if total_count > MIN_ROWS_LOG_BATCH and (row_count % batch_percent) == 0:
-                        percent = math.floor((row_count/total_count)*100)
-                        log_data['message'] = 'Completed derivations for {} rows ({}%)'.format(row_count, percent)
-                        logger.info(log_data)
-                    row_count += 1
-                loop_num += 1
-
-            # update all cached D2 FileGeneration objects that could have been affected by the publish
-            sess.query(FileGeneration).\
-                filter(FileGeneration.agency_code.in_(agency_codes_list),
-                       FileGeneration.is_cached_file.is_(True),
-                       FileGeneration.file_type == 'D2').\
-                update({'is_cached_file': False}, synchronize_session=False)
+            log_derivation('Starting derivations for FABS submission (total count: {})'.format(total_count),
+                           submission_id)
+
+            # Insert all non-error, non-delete rows into published table
+            column_list = [col.key for col in DetachedAwardFinancialAssistance.__table__.columns]
+            column_list.remove('created_at')
+            column_list.remove('updated_at')
+            column_list.remove('detached_award_financial_assistance_id')
+            column_list.remove('job_id')
+            column_list.remove('row_number')
+            column_list.remove('is_valid')
+            detached_col_string = ", ".join(column_list)
+
+            column_list = [col.key for col in PublishedAwardFinancialAssistance.__table__.columns]
+            column_list.remove('created_at')
+            column_list.remove('updated_at')
+            column_list.remove('modified_at')
+            column_list.remove('is_active')
+            column_list.remove('published_award_financial_assistance_id')
+            published_col_string = ", ".join(column_list)
```

Comment:
Could we clean this up like (or even using python's filters works too) 
```suggestion
            column_list = [col.key for col in DetachedAwardFinancialAssistance.__table__.columns]
            remove_cols = ['created_at', 'updated_at', 'detached_award_financial_assistance_id', 'job_id', 'row_number', 'is_valid']
            for remove_col in remove_cols:
                column_list.remove(remove_col)
            detached_col_string = ", ".join(column_list)

            column_list = [col.key for col in PublishedAwardFinancialAssistance.__table__.columns]
            remove_cols = ['created_at', 'updated_at', 'modified_at', 'published_award_financial_assistance_id']
            for remove_col in remove_cols:
                column_list.remove(remove_col)
            published_col_string = ", ".join(column_list)
```